flight/Modules/ManualControl/transmitter_control.c

```diff
@@ -490,139 +478,205 @@ static void set_armed_if_changed(uint8_t new_arm) {
+			(settings->ArmTime == MANUALCONTROLSETTINGS_ARMTIME_500) ? 500 : \
+			(settings->ArmTime == MANUALCONTROLSETTINGS_ARMTIME_1000) ? 1000 : \
+			(settings->ArmTime == MANUALCONTROLSETTINGS_ARMTIME_2000) ? 2000 : 1000;
+		if (arm && timeDifferenceMs(armedDisarmStart, lastSysTime) > arm_time) {
+			arm_state = ARM_STATE_ARMED_STILL_HOLDING;
+		} else if (!arm) {
+			arm_state = ARM_STATE_DISARMED;
 		}
+	}
+		break;
 
-		bool manualArm = false;
-		bool manualDisarm = false;
-
-		if (armingInputLevel <= -ARMED_THRESHOLD)
-			manualArm = true;
-		else if (armingInputLevel >= +ARMED_THRESHOLD)
-			manualDisarm = true;
-
-		switch(arm_state) {
-			case ARM_STATE_DISARMED:
-				set_armed_if_changed(FLIGHTSTATUS_ARMED_DISARMED);
+	case ARM_STATE_ARMED_STILL_HOLDING:
+	{
+		set_armed_if_changed(FLIGHTSTATUS_ARMED_ARMED);
 
-				if (manualArm) {
-					armedDisarmStart = lastSysTime;
-					arm_state = ARM_STATE_ARMING_MANUAL;
-				}
+		bool arm = arming_position(cmd, settings) && valid;
+		if (!arm) {
+			arm_state = ARM_STATE_ARMED;
+		}
+	}
+		break;
 
-				break;
+	case ARM_STATE_ARMED:
+	{
+		set_armed_if_changed(FLIGHTSTATUS_ARMED_ARMED);
 
-			case ARM_STATE_ARMING_MANUAL:
-				set_armed_if_changed(FLIGHTSTATUS_ARMED_ARMING);
+		// Check for arming timeout if transmitter invalid
+		if (!valid) {
+			if ((settings->ArmedTimeout != 0) && (timeDifferenceMs(armedDisarmStart, lastSysTime) > settings->ArmedTimeout))
+				arm_state = ARM_STATE_DISARMED;
+		} else {
+			armedDisarmStart = lastSysTime;
+ 		}
 
-				if (manualArm && (timeDifferenceMs(armedDisarmStart, lastSysTime) > ARMED_TIME_MS))
-					arm_state = ARM_STATE_ARMED;
-				else if (!manualArm)
-					arm_state = ARM_STATE_DISARMED;
-				break;
+  		bool disarm = disarming_position(cmd, settings) && valid;
+		if (disarm && (settings->Arming == MANUALCONTROLSETTINGS_ARMING_SWITCH ||
+		       settings->Arming == MANUALCONTROLSETTINGS_ARMING_SWITCHTHROTTLE)) {
+			arm_state = ARM_STATE_DISARMED;
+		} else if (disarm) {
+			armedDisarmStart = lastSysTime;
+			arm_state = ARM_STATE_DISARMING;
+		}
+	}
+  		break;
 
-			case ARM_STATE_ARMED:
-				// When we get here, the throttle is low,
-				// we go immediately to disarming due to timeout, also when the disarming mechanism is not enabled
-				armedDisarmStart = lastSysTime;
-				arm_state = ARM_STATE_DISARMING_TIMEOUT;
-				set_armed_if_changed(FLIGHTSTATUS_ARMED_ARMED);
-				break;
+	case ARM_STATE_DISARMING:
+	{
+  		bool disarm = disarming_position(cmd, settings) && valid;
+		uint16_t disarm_time = (settings->DisarmTime == MANUALCONTROLSETTINGS_DISARMTIME_250) ? 250 : \
+			(settings->DisarmTime == MANUALCONTROLSETTINGS_DISARMTIME_500) ? 500 : \
+			(settings->DisarmTime == MANUALCONTROLSETTINGS_DISARMTIME_1000) ? 1000 : \
+			(settings->DisarmTime == MANUALCONTROLSETTINGS_DISARMTIME_2000) ? 2000 : 1000;
+  		if (disarm && timeDifferenceMs(armedDisarmStart, lastSysTime) > disarm_time) {
+			arm_state = ARM_STATE_DISARMING_STILL_HOLDING;
```

Comment:
the behavior as it goes to the next state is to immediately disarm (see line 670). however, it has to stay there so if they hold the sticks it doesn't enter the disarmed state and then immediately start arming again.

I've renamed ```ARM_STATE_DISARMING_STILL_HOLDING``` to ```ARM_STATE_DISARMED_STILL_HOLDING``` to make it clear this is already disarmed.